Duplicate credential isn't deleted on same user handle when requested attestation is not none

pascoej opened the pull request:
#### 5ab8984adac5197eef752a7b6a82509db3efd3fe
<pre>
Duplicate credential isn&apos;t deleted on same user handle when requested attestation is not none
<a href="https://bugs.webkit.org/show_bug.cgi?id=261349">https://bugs.webkit.org/show_bug.cgi?id=261349</a>
rdar://problem/115188953

Reviewed by Brent Fulgham.

Previously, we would error out a request if we couldn&apos;t satisfy a given
attestation type for our platform authenticator. However, after our platform
authenticator stopped supporting attestation, we changed the behavior to
return a none attestation, but omitted the step of deleting any duplicate
credentials with the same user handle. This patch fixes that by deleting
the potential duplicate credential.

* Source/WebKit/UIProcess/WebAuthentication/Cocoa/LocalAuthenticator.mm

Canonical link: <a href="https://commits.webkit.org/269309@main">https://commits.webkit.org/269309@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/432b21db8b62d68422f5ced9cd26e2a6fc5b9bce

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/17758 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/18088 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/18626 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/19582 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/16611 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/21379 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/18238 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/18660 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/17971 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/18261 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/15430 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/20478 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/15493 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/22762 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/16505 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/16351 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/20626 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/16919 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/14325 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/16032 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5282 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/20395 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/16764 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham reviewed (approved):
Is there any other code we can remove if the Attestation logic is no longer used?

pascoej committed:
Duplicate credential isn't deleted on same user handle when requested attestation is not none
https://bugs.webkit.org/show_bug.cgi?id=261349
rdar://problem/115188953

Reviewed by Brent Fulgham.

Previously, we would error out a request if we couldn't satisfy a given
attestation type for our platform authenticator. However, after our platform
authenticator stopped supporting attestation, we changed the behavior to
return a none attestation, but omitted the step of deleting any duplicate
credentials with the same user handle. This patch fixes that by deleting
the potential duplicate credential.

* Source/WebKit/UIProcess/WebAuthentication/Cocoa/LocalAuthenticator.mm

Canonical link: https://commits.webkit.org/269309@main
